run/multibit2john.py

```diff
@@ -17,1282 +17,10 @@
-  DESCRIPTOR = _TRANSACTIONINPUT,
-  __module__ = 'wallet_pb2'
-  # @@protoc_insertion_point(class_scope:wallet.TransactionInput)
-  ))
-_sym_db.RegisterMessage(TransactionInput)
-
-TransactionOutput = _reflection.GeneratedProtocolMessageType('TransactionOutput', (_message.Message,), dict(
-  DESCRIPTOR = _TRANSACTIONOUTPUT,
-  __module__ = 'wallet_pb2'
-  # @@protoc_insertion_point(class_scope:wallet.TransactionOutput)
-  ))
-_sym_db.RegisterMessage(TransactionOutput)
-
-TransactionConfidence = _reflection.GeneratedProtocolMessageType('TransactionConfidence', (_message.Message,), dict(
-  DESCRIPTOR = _TRANSACTIONCONFIDENCE,
-  __module__ = 'wallet_pb2'
-  # @@protoc_insertion_point(class_scope:wallet.TransactionConfidence)
-  ))
-_sym_db.RegisterMessage(TransactionConfidence)
-
-Transaction = _reflection.GeneratedProtocolMessageType('Transaction', (_message.Message,), dict(
-  DESCRIPTOR = _TRANSACTION,
-  __module__ = 'wallet_pb2'
-  # @@protoc_insertion_point(class_scope:wallet.Transaction)
-  ))
-_sym_db.RegisterMessage(Transaction)
-
-ScryptParameters = _reflection.GeneratedProtocolMessageType('ScryptParameters', (_message.Message,), dict(
-  DESCRIPTOR = _SCRYPTPARAMETERS,
-  __module__ = 'wallet_pb2'
-  # @@protoc_insertion_point(class_scope:wallet.ScryptParameters)
-  ))
-_sym_db.RegisterMessage(ScryptParameters)
-
-Extension = _reflection.GeneratedProtocolMessageType('Extension', (_message.Message,), dict(
-  DESCRIPTOR = _EXTENSION,
-  __module__ = 'wallet_pb2'
-  # @@protoc_insertion_point(class_scope:wallet.Extension)
-  ))
-_sym_db.RegisterMessage(Extension)
-
-Tag = _reflection.GeneratedProtocolMessageType('Tag', (_message.Message,), dict(
-  DESCRIPTOR = _TAG,
-  __module__ = 'wallet_pb2'
-  # @@protoc_insertion_point(class_scope:wallet.Tag)
-  ))
-_sym_db.RegisterMessage(Tag)
-
-TransactionSigner = _reflection.GeneratedProtocolMessageType('TransactionSigner', (_message.Message,), dict(
-  DESCRIPTOR = _TRANSACTIONSIGNER,
-  __module__ = 'wallet_pb2'
-  # @@protoc_insertion_point(class_scope:wallet.TransactionSigner)
-  ))
-_sym_db.RegisterMessage(TransactionSigner)
-
-Wallet = _reflection.GeneratedProtocolMessageType('Wallet', (_message.Message,), dict(
-  DESCRIPTOR = _WALLET,
-  __module__ = 'wallet_pb2'
-  # @@protoc_insertion_point(class_scope:wallet.Wallet)
-  ))
-_sym_db.RegisterMessage(Wallet)
-
-ExchangeRate = _reflection.GeneratedProtocolMessageType('ExchangeRate', (_message.Message,), dict(
-  DESCRIPTOR = _EXCHANGERATE,
-  __module__ = 'wallet_pb2'
-  # @@protoc_insertion_point(class_scope:wallet.ExchangeRate)
-  ))
-_sym_db.RegisterMessage(ExchangeRate)
-
-
-DESCRIPTOR.has_options = True
-DESCRIPTOR._options = _descriptor._ParseOptions(descriptor_pb2.FileOptions(), _b('\n\023org.bitcoinj.walletB\006Protos'))
-# @@protoc_insertion_point(module_scope)
-
-# End of automatically generated code.
+# The precompiled file is shipped in protobuf/wallet_pb2.py with John.
+from protobuf.wallet_pb2 import *
```

Comment:
I'm not familiar with how Python resolves `from protobuf.wallet_pb2 import *` into script file to import. If the main script is run as e.g. `/path/to/john/run/multibit2john.py`, what directories would be checked for the imported script? Would it be only `/path/to/john/run/protobuf` or perhaps also some others - perhaps at least some system directories? Would this include the current directory (a security concern)?

Reply:
It prefers the location of the script file, i.e. John's `run/` directory. So it should be safe.